tools/generator-go-sdk/generator/templater_methods_autorest.go

```diff
@@ -653,6 +652,66 @@ func (c %[1]s) responderFor%[2]s(resp *http.Response) (result %[6]s, err error)
 		return &output, nil
 	}
 
+	if discriminatedType != "" && strings.HasPrefix(c.operationName, "List") {
+		typeName, err := golangTypeNameForObjectDefinition(*c.operation.ResponseObject)
+		if err != nil {
+			return nil, fmt.Errorf("determining golang type name for response object: %+v", err)
+		}
+		fields := make([]string, 0)
+		fields = append(fields, fmt.Sprintf("Values []json.RawMessage `json:%q`", "value"))
+		fields = append(fields, fmt.Sprintf("NextLink *string `json:%q`", *c.operation.FieldContainingPaginationDetails))
+		output := fmt.Sprintf(`
+// responderFor%[2]s handles the response to the %[2]s request. The method always
+// closes the http.Response Body.
+func (c %[1]s) responderFor%[2]s(resp *http.Response) (result %[6]s, err error) {
+	type page struct {
+		%[4]s
+	}
+	var respObj page
+	err = autorest.Respond(
+		resp,
+		%[3]s)
+	result.HttpResponse = resp
+	temp := make([]%[7]s, 0)
+	for _, v := range respObj.Values {
+		val, err := unmarshal%[8]sImplementation(v)
+			if err != nil {
+				err = autorest.NewErrorWithError(err, "%[5]s.%[1]s", "%[2]s", nil, "Failure to unmarshal")
```

Comment:
it'd be good to give details for /which/ iteration we've had an unmarshalling error with:

```suggestion
	for i, v := range respObj.Values {
		val, err := unmarshal%[8]sImplementation(v)
			if err != nil {
				err = fmt.Errorf("unmarshalling item %%d for %[8]s (%%q): %+v", i, v, err)
```